src/librustdoc/clean/mod.rs

```diff
@@ -2231,6 +2231,12 @@ impl Clean<Vec<Item>> for doctree::ExternCrate<'_> {
 
 impl Clean<Vec<Item>> for doctree::Import<'_> {
     fn clean(&self, cx: &DocContext<'_>) -> Vec<Item> {
+        // We need this comparison because some imports (for std types for example)
+        // are "inserted" as well but directly by the compiler and they should not be
+        // taken into account.
+        if self.span.is_dummy() {
+            return Vec::new();
+        }
```

Comment:
We can't test it. The compiler inserts some imports like `std` for example. So in case we receive an import with a dummy span, it means that it was inserted by the compiler and that we don't want to check it.

Reply:
Guillaume said that otherwise this includes the injected `use std::prelude::v1::*;` imports. I agree a test would be nice to have.

Reply:
> We can't test it

@GuillaumeGomez you could test that the item doesn't appear with `@has`. Or whatever issue you ran into originally that made you add this check.

Reply:
The issue I had was simply that a weird error message appeared with missing doc example (when I was forcing the check to test it) with a span pointing to nowhere. So literally: we can't test it.

Reply:
`use std::prelude::v1::*;` imports are never inlined so I don't see what that has to do with this PR which is only changing behavior for imports that are inlined.

> The issue I had was simply that a weird error message appeared with missing doc example (when I was forcing the check to test it) with a span pointing to nowhere. So literally: we can't test it.

If you can't write a `rustdoc-ui` test or at least provide instructions on how to reproduce this issue then this code needs to be deleted.

Reply:
To trigger it, you need to unable a check to be run on the import item doc comments from rustdoc, like `missing_doc_code_example`. Then you'll have a nice error showing up first, coming out from nowhere. So no, this code is necessary. Maybe we don't have the use for it right now, but who knows if we won't add a lint that will need to check import items later on? Then the bug would reappear and it would make the contributor lose time for something that could have been fixed earlier on. Dummy imports are added by the compiler and shouldn't be considered as rustdoc items and therefore should be ignored.

Reply:
Okay, I agree that rustdoc should ignore imports automatically added by rustc. However, that doesn't seem to have anything to do with the rest of this PR so it should be moved to a separate one although this could just be done if/when a lint is added that would actually need this.

Reply:
Unfortunately, we add back the imports as items to be checked by the different passes. We need to filter them out in the same PR, otherwise it'll be problematic. So if not in this PR, I wonder we can do it. :-/

Reply:
The only `ImportItem`s this PR is adding are ones the have been inlined. The prelude imports are private and so are never inlined so this PR is not adding new `ImportItem`s for them.

There is actually a visible change from this. Currently, if you build docs with `--no-defaults` then `use std::prelude::v1::*;` shows up in the Re-exports section in the crate root. This change removes that line. That's something that could be tested assuming hiding that line is intentional. `extern crate std;` also shows up, should that also be hidden?

Again, please could you move this change to a different PR.

Reply:
Ok, that seems fair. I opened https://github.com/rust-lang/rust/issues/77567.